Time-Card/TOOL/main.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"bytes"
+	"encoding/binary"
+	"flag"
+	"fmt"
+	"os"
+
+	crc "github.com/sigurn/crc16"
+	log "github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	Apply            bool
+	InputPath        string
+	OutputPath       string
+	VendorId         int
+	DeviceId         int
+	HardwareRevision int
+}
+
+type Header struct {
+	Magic            [4]byte
+	VendorId         uint16
+	DeviceId         uint16
+	ImageSize        uint32
+	HardwareRevision uint16
+	CRC              uint16
+}
+
+func main() {
```

Comment:
Ideally main should only have flag/library call and the rest should go to ./header/*.go